tests/Unit/Domain/CoordinateMaps/TimeDependent/Test_SphericalCompression.cpp

```diff
@@ -0,0 +1,595 @@
+  const double initial_time{time - dt_dis(*generator)};
+  const double expiration_time{time + dt_dis(*generator)};
+
+  const std::string f_of_t_name{"ExpansionFactor"};
+  functions_of_time[f_of_t_name] =
+      std::make_unique<domain::FunctionsOfTime::PiecewisePolynomial<3>>(
+          initial_time, initial_coefficients, expiration_time);
+  CHECK_FALSE(static_cast<bool>(
+      map.inverse(map(point(radius_dis(*generator)), time, functions_of_time),
+                  time, functions_of_time)));
+  functions_of_time[f_of_t_name] =
+      std::make_unique<domain::FunctionsOfTime::PiecewisePolynomial<3>>(
+          initial_time, initial_coefficients_b, expiration_time);
+  CHECK_FALSE(static_cast<bool>(
+      map.inverse(map(point(radius_dis(*generator)), time, functions_of_time),
+                  time, functions_of_time)));
+}
+
+template <bool InteriorMap>
+void test_is_identity(gsl::not_null<std::mt19937*> generator) noexcept {
+  INFO("Is identity");
+  CoordinateMaps::TimeDependent::SphericalCompression<InteriorMap> map{};
+  double time{std::numeric_limits<double>::signaling_NaN()};
+  std::unordered_map<std::string,
+                     std::unique_ptr<::domain::FunctionsOfTime::FunctionOfTime>>
+      functions_of_time{};
+  generate_map_time_and_f_of_time(make_not_null(&map), make_not_null(&time),
+                                  make_not_null(&functions_of_time), generator);
+  CHECK(not map.is_identity());
+}
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Domain.CoordinateMaps.SphericalCompression",
+                  "[Domain][Unit]") {
+  MAKE_GENERATOR(gen);
+  test_suite<true>(make_not_null(&gen));
+  test_map<true>(make_not_null(&gen));
+  test_is_identity<true>(make_not_null(&gen));
+  test_suite<false>(make_not_null(&gen));
+  test_map<false>(make_not_null(&gen));
+  test_is_identity<false>(make_not_null(&gen));
+}
+
+// [[OutputRegex, max_radius must be greater]]
+[[noreturn]] SPECTRE_TEST_CASE(
+    "Unit.Domain.CoordinateMaps.SphericalCompression.MaxGreater",
+    "[Domain][Unit]") {
+  ASSERTION_TEST();
+  MAKE_GENERATOR(gen);
+  CoordinateMaps::TimeDependent::SphericalCompression<false> map{};
+  double time{std::numeric_limits<double>::signaling_NaN()};
+  std::unordered_map<std::string,
+                     std::unique_ptr<::domain::FunctionsOfTime::FunctionOfTime>>
+      functions_of_time{};
+  generate_map_time_and_f_of_time(make_not_null(&map), make_not_null(&time),
+                                  make_not_null(&functions_of_time),
+                                  make_not_null(&gen), true, false);
+  test_suite<false>(make_not_null(&gen));
+  // should never be called, but suppresses a warning that noreturn functions
+  // should not return
+  std::terminate();
+}
+
+// [[OutputRegex, Could not find function of time]]
+[[noreturn]] SPECTRE_TEST_CASE(
+    "Unit.Domain.CoordinateMaps.SphericalCompression.MissingFofT",
+    "[Domain][Unit]") {
+  ASSERTION_TEST();
+  MAKE_GENERATOR(gen);
```

Comment:
Most of our assertion tests follow the pattern of wrapping their contents in a `#ifdef SPECTRE_DEBUG ... #endif`, and having an `ERROR("Failed to trigger ASSERT in an assertion test)`, I think to avoid the need for the `std::terminate()` at the end -- is there a reason for preferring to not use that pattern here?
(if you change this assertion test, then also propagate a similar change to the previous assertion test.)

Reply:
I also fixed the ERROR tests similarly, removing std::terminate() and instead having a "failed to trigger ERROR" call to ERROR() at the end